presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java

```diff
@@ -3216,10 +3241,14 @@ protected void doCreateTable(SchemaTableName tableName, HiveStorageFormat storag
             ConnectorOutputTableHandle outputHandle = metadata.beginCreateTable(session, tableMetadata, Optional.empty());
 
             // write the data
-            ConnectorPageSink sink = pageSinkProvider.createPageSink(transaction.getTransactionHandle(), session, outputHandle, PageSinkProperties.defaultProperties());
+            ConnectorPageSink sink = pageSinkProvider.createPageSink(transaction.getTransactionHandle(), session, outputHandle, pageSinkProperties);
             sink.appendPage(CREATE_TABLE_DATA.toPage());
             Collection<Slice> fragments = getFutureValue(sink.finish());
 
+            if (pageSinkProperties.isPartitionCommitRequired()) {
+                metadata.commitPartitions(session, outputHandle, fragments);
```

Comment:
Let's also decode the `fragments` and check that the writePath and the targetPath are different if commit is required